libpromises/cf3parse.y

```diff
@@ -755,10 +755,21 @@ litem:                 IDENTIFIER
                                        (P.currentclasses ?
                                             P.currentclasses : "any"),
                                        P.currentstring);
-                           RlistAppendScalar((Rlist **) &P.currentRlist,
-                                             (void *) P.currentstring);
-                           free(P.currentstring);
-                           P.currentstring = NULL;
+
+                           if (!StringEqual(P.lval + 1, "list") || P.lval[0] == 's')  // slists and everything else
```

Comment:
This code assumes `!NULL_OR_EMPTY(P.lval);`. Please add another if or assert for this.